File: 283-move-zeroes/283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        for(int i=nums.length-1; i>=0; i--){
            for(int j=1; j<=i; j++){
                if(nums[j-1] == 0){
                    int temp = nums[j-1];
                    nums[j-1] = nums[j];
                    nums[j] = temp;
                }
            }
        }
    }
}

File: 283-move-zeroes/283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i{0};
        int j{0};
        while(i < nums.size() && j < nums.size()){
            if(nums.at(i) == 0 && nums.at(j) != 0){
                int temp = nums.at(i);
                nums.at(i) = nums.at(j);
                nums.at(j) = temp;
                i++;
                j++;
            }
            else if(nums.at(i) != 0 && nums.at(j) == 0){
                i++;
            }
            else {
                j++;
            }
        }
    }
};